packages/components/forms/src/FormControl/FormControlContext.tsx

```diff
@@ -0,0 +1,52 @@
+import { createContext, useContext } from 'react';
+
+export interface FormControlProps {
+  id?: string;
+  /**
+   * If `true` set the form control to the invalid state.
+   */
+  isInvalid?: boolean;
+  /**
+   * If `true` set the form control to be required.
+   */
+  isRequired?: boolean;
+  /**
+   * If `true` set the form control to the disabled state.
+   */
+  isDisabled?: boolean;
+
+  isReadOnly?: boolean;
+}
+
+export const FormControlContext = createContext<FormControlProps | undefined>(
+  undefined,
+);
```

Comment:
We prefer `undefined` over `null`?

is that why you passed `undefined` to `createContext`?

Reply:
I don't really see the difference between these two here.

Reply:
Sorry, I was a bit confused

my question should be more: do we want the initial value to be undefined? Could it be an empty object perhaps?

so when I do:

```tsx
const { isInvalid } = useFormControlContext() // isInvalid will be initially undefined but this won't break the app
```

Reply:
yeah, the ones from context should ideally be undefined unless there's a specific value set on `FormControl`